File: src/main/java/com/pedigree/model/Note.java
package com.pedigree.model;

import java.util.List;
import java.util.UUID;

public class Note {
    private final String id = UUID.randomUUID().toString();
    private String text;
    private final List<SourceCitation> sources = new com.pedigree.util.DirtyObservableList<>();

    public String getId() { return id; }
    public String getText() { return text; }
    public void setText(String text) {
        this.text = text;
        com.pedigree.util.DirtyFlag.setModified();
    }
    public List<SourceCitation> getSources() { return sources; }
}

File: src/main/kotlin/com/family/tree/model/Note.kt
package com.family.tree.model

import com.family.tree.util.DirtyFlag
import com.family.tree.util.DirtyObservableList
import java.util.UUID

class Note {
    val id: String = UUID.randomUUID().toString()
    var text: String? = null
        set(value) {
            field = value
            DirtyFlag.setModified()
        }
    val sources: MutableList<SourceCitation> = DirtyObservableList()
}
